# Projet Fil Rouge Centrale #

Victor Bonlieu-Fortier, Charles Nikdel

---

## Importation librairies ##

In [56]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import utils as ft_nous


In [25]:
rename_columns = [
					"index",
					"date_urg",
					"conduite",
					"m_entry",
					"m_venue",
					"age",
					"atcd_med",
					"atcd_chir",
					"cst_paspad",
					"fc_ad",
					"tmp_ad",
					"saoxy_ad",
					"fr_ad",
					"ccmu",
					"salle",
					"date_box",
					"anamn",
					"obs",
					"have_bio",
					"have_radio",
					"have_echo",
					"have_scan",
					"have_irm",
					"have_specia",
					"date_prescri_bio",
					"date_prelev_bio",
					"code_diag",
					"intitule_diag",
					"cac_admi",
					"cac_sejour",
					"date_sortie",
					"date_deb_pec_med",
					"dossier_hopit",
					"type_orient",
					"destination",
					"transfert_service",
					"transfert_hopit",
					"cp_ville",
					"ville"
					]
rename_columns


['index',
 'date_urg',
 'conduite',
 'm_entry',
 'm_venue',
 'age',
 'atcd_med',
 'atcd_chir',
 'cst_paspad',
 'fc_ad',
 'tmp_ad',
 'saoxy_ad',
 'fr_ad',
 'ccmu',
 'salle',
 'date_box',
 'anamn',
 'obs',
 'have_bio',
 'have_radio',
 'have_echo',
 'have_scan',
 'have_irm',
 'have_specia',
 'date_prescri_bio',
 'date_prelev_bio',
 'code_diag',
 'intitule_diag',
 'cac_admi',
 'cac_sejour',
 'date_sortie',
 'date_deb_pec_med',
 'dossier_hopit',
 'type_orient',
 'destination',
 'transfert_service',
 'transfert_hopit',
 'cp_ville',
 'ville']

In [26]:
df = pd.read_excel("../database/db_urg.xlsx", index_col=0, header=0, names = rename_columns)
df


date_urg           conduite  \
index                                           
1      2016-06-01 00:16:00   AMBULANCE PRIVEE   
2      2016-06-01 00:18:00   AMBULANCE PRIVEE   
3      2016-06-01 00:36:00           POMPIERS   
4      2016-06-01 00:43:00  MOYENS PERSONNELS   
5      2016-06-01 01:16:00   AMBULANCE PRIVEE   
...                    ...                ...   
336249 2020-06-01 22:58:00   AMBULANCE PRIVEE   
336250 2020-06-01 23:03:00   AMBULANCE PRIVEE   
336251 2020-06-01 23:04:00   AMBULANCE PRIVEE   
336252 2020-06-01 23:12:00  MOYENS PERSONNELS   
336253 2020-06-01 23:54:00  MOYENS PERSONNELS   

                                                  m_entry  \
index                                                       
1       Malaise, difficulté respi, palpitation, Pas de...   
2       AEG, Hyperthermie et difficulté respi en maiso...   
3                                                     NaN   
4       1A, Hyperthermie ne cédant pas au paracetamol,...   
5       Douleur bras gauche et mollet gauche, hémiplég...   
...                                                   ...   
336249  Suspicion  de pyélonéphrite,  passage au SAU  ...   
336250     Hyperglycémie à 7gr au domicile , 0.98 en  ZAO   
336251                     Suspicion d'infection urinaire   
336252                                        Douleur FIG   
336253                           Plaie derrière oreille G   

                                                  m_venue  age  \
index                                                            
1                       Cardiologie: Palpitations Malaise   33   
2                     Maladies infectieuses: Hyperthermie   78   
3                                                     NaN   77   
4                     Maladies infectieuses: Hyperthermie   38   
5                                   Cardiologie: Phlébite   69   
...                                                   ...  ...   
336249        Uro-néphrologie: Suspicion de pyélonéphrite   49   
336250        Endocrino-métabolisme: Déséquilibre diabète   64   
336251  Uro-néphrologie: Dysurie, brulure mictionnelle...   72   
336252  Gastro-entérologie: Douleur abdominale non épi...   16   
336253                 Traumatologie: Plaie superficielle   63   

                                                 atcd_med           atcd_chir  \
index                                                                           
1                                                     NaN                 NaN   
2                                                     NaN                 NaN   
3                                                     NaN                 NaN   
4                                                     NaN                 NaN   
5       décembre 2007 : infarctus capsulo caudé droit ...                 NaN   
...                                                   ...                 ...   
336249                                                ras                 NaN   
336250                                                NaN                 NaN   
336251                  ESV paroxystiques   Fissure anale  Hernie de l'aine     
336252                                                NaN                 NaN   
336253                                                NaN                 NaN   

       cst_paspad fc_ad tmp_ad  ... cac_sejour   date_sortie date_deb_pec_med  \
index                           ...                                             
1          110/56    52   34.9  ...     3093.0  42522.176389     42522.043750   
2           90/41    77   36.3  ...     3102.0  42522.461806     42522.093056   
3          137/77    97   36.5  ...     3102.0  42522.127778              NaN   
4          117/71   118   38.2  ...     3093.0  42522.201389     42522.081250   
5          137/79    70   36.1  ...     3102.0  42522.663194     42522.107639   
...           ...   ...    ...  ...        ...           ...              ...   
336249     101/71    93  

In [32]:
df.info()


<class 'pandas.core.frame.DataFrame'>
Index: 336253 entries, 1 to 336253
Data columns (total 38 columns):
 #   Column             Non-Null Count   Dtype         
---  ------             --------------   -----         
 0   date_urg           336253 non-null  datetime64[ns]
 1   conduite           334907 non-null  object        
 2   m_entry            311802 non-null  object        
 3   m_venue            307534 non-null  object        
 4   age                336253 non-null  int64         
 5   atcd_med           102623 non-null  object        
 6   atcd_chir          71134 non-null   object        
 7   cst_paspad         242618 non-null  object        
 8   fc_ad              244278 non-null  object        
 9   tmp_ad             267414 non-null  object        
 10  saoxy_ad           225761 non-null  object        
 11  fr_ad              10785 non-null   object        
 12  ccmu               293122 non-null  object        
 13  salle              336253 non-null  object       

On voit que bon, y a beaucoup de colonne mais pas mal qui contienne rien d'interessant, on va donc les supprimer
Dans l'ordre ceux que je pense supprimer et concatenner pour certaines : 

- obs sert pas a grand chose et y en a pas bcq, c'est les observations des inf genre EVA ou autre
- code diag donne le code du diagnostic, on a deja le libelle donc pas besoin
- cac admi jsp vraiment ce que c'est ca doit etre un code de protocole, mais il faut juste récup le numéro comme le cac_sejour
- dossier_hopit y es casi nul donc on peut tej
- type orientation, c'est destination mais moins précis, on peut tej
- transfert service c'est destination mais juste pour les services internes, on peut tej
- transfert etab c'est destination mais juste pour les autres etablissements, y en as pas enorme, je pense on peut tej et et juste modif pour dire autre etab
- le code postal et la ville jsp si c'est utile la vérité à voir, je pense qu'on peut retirer le code postal et garder la ville pour des vizu de zinzin
- frequence respi y en as pas bcq on peut retirer
- Je pense qu'on peut retirer aussi tout ceux qui n'ont pas un code ccmu, ca veut dire ceux qui ont pas été triagé et donc pas pris en charge par les urgences
- salle enfaite c'est une liste de leur voyage dans les urgences, jsp si c'est utile non plus , ca clc à parser et jsp si ca sera utile

Pour réduire le nbr de colonne et le contenu en info :

- Conduite, on peut changer ca en 1 2 3 4 5 ... y 'en a pas bcq
- motif entrée et admission : clean et tokenizer comme on as fait sur twitter
- atcd med et chir idem 
- les constantes jsp pq aucune est en int il faut convert : trouver un truc pck la pression sanguine en 110 / 56 ca clc par exemple vu que c'est pas un int
- convertir la température en int, pour remplacer les NaN juste mettre la température normale
- la saturation en oxy, pareil remplacer les NaN par la saturation normale et convertir en int sachant que y a des inf clc qui ont met AA (air ambiant) du coup les retirer
- anamnèse, on peut tokenizer et clean comme on as fait sur twitter
- tout les have, à transformer en 0 1
- les date prescri bio et prelev, peut être juste faire un delta entre les deux, y en as ils sont pas prescri donc on peut mettre 0 et d'autre qui sont prescri mais pas prelevé donc mettre la date de prescri ou la moyenne des dates de prescri jpense
- intitule diag à tokenizer et clean
- parser destination pour juste garder le service / hopital / autre hopital
- ville ca a l'air d'être bon

In [42]:
df.columns


Index(['date_urg', 'conduite', 'm_entry', 'm_venue', 'age', 'atcd_med',
       'atcd_chir', 'cst_paspad', 'fc_ad', 'tmp_ad', 'saoxy_ad', 'fr_ad',
       'ccmu', 'salle', 'date_box', 'anamn', 'obs', 'have_bio', 'have_radio',
       'have_echo', 'have_scan', 'have_irm', 'have_specia', 'date_prescri_bio',
       'date_prelev_bio', 'code_diag', 'intitule_diag', 'cac_admi',
       'cac_sejour', 'date_sortie', 'date_deb_pec_med', 'dossier_hopit',
       'type_orient', 'destination', 'transfert_service', 'transfert_hopit',
       'cp_ville', 'ville'],
      dtype='object')

In [ ]:
df = df.dropna(subset=["ccmu"])
df = df.drop(["code_diag","dossier_hopit","type_orient","transfert_service","transfert_hopit","fr_ad"],axis=1)
df.info()


In [ ]:
df = df.drop(["obs"],axis=1)
df


In [86]:
df["conduite"].values[233]


'AMB PUBLIQUE CHRU '

Maintenant je vais faire des fonctions pour traiter les données et les remettre dans le bon type en général

In [89]:
df = ft_nous.convertir_conduite(df, "conduite")
df["conduite"].value_counts()


ValueError: invalid literal for int() with base 10: 'AMB PUBLIQUE CHRU '